src/ips/algorithm.rs

```diff
@@ -128,11 +165,23 @@ impl Ips {
                 // after separation.
                 panic!("There are more than one massive island in the network. It is not possible to merge them automatically.");
```

Comment:
I haven't seen this change until now - nice!
(_this massive_count block check_)